src/message.c

```diff
@@ -1580,7 +1580,7 @@ qd_message_t *qd_message_receive(pn_delivery_t *delivery)
                     // pending buffer is absent
                 }
 
-                content->receive_complete = true;
+                SET_ATOMIC_BOOL(&content->receive_complete, true);
```

Comment:
Does this clear any `tsan.supp` entry, by any chance?

I noticed

https://github.com/skupperproject/skupper-router/blob/81a196b1c94de870a0f8d85eb77a792c6ab94d76/tests/tsan.supp#L35-L36

but looking at https://issues.apache.org/jira/browse/DISPATCH-2127, it does appear that atomically setting this flag will allow getting rid of the suppression.

Reply:
Thanks @jiridanek , I will take a look

Reply:
The removal of the suppression is dependent upon these three flags being accessed safely -
```
content->receive_complete
content->q2_input_holdoff
content->aborted
```

I have added additional code to this PR so that access to content->q2_input_holdoff is secured after holding `content->lock`

Let's see what the CI thinks.

Reply:
After removing the suppression, I see no new test failures on this PR. I am going to kick off the tests one more time to see what happens.

Reply:
Removing the suppression did not have any side effects. @jiridanek @kgiusti please take a look at this PR one more time and approve/comment.